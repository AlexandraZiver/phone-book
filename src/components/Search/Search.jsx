import PropTypes from "prop-types";
import { Input } from "semantic-ui-react";

const Search = ({ value, onChange }) => (
  <Input
    type="text"
    value={value}
    name="searchInput"
    onChange={onChange}
    icon="search"
    placeholder="Search..."
  />
);

Search.defaultProps = {
  value: "",
};

Search.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default Search;
