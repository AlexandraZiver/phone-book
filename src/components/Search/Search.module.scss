@import "../../styles/theme";
.Input {
   background-color: theme-color(secondary);
}
