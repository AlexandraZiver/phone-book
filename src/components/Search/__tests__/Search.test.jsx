import { fireEvent, render, screen } from "@testing-library/react";

import Search from "../index";

import { Input } from "semantic-ui-react";

jest.mock("semantic-ui-react", () => {
  return {
    Input: jest
      .fn()
      .mockImplementation(({ value, onChange }) => (
        <input
          type="text"
          value={value || ""}
          name="searchInput"
          onChange={onChange || jest.fn()}
          icon="search"
          placeholder="Search..."></input>
      )),
  };
});
describe("testing Search ", () => {
  test("testing by onChange of search", () => {
    const handleChange = jest.fn();

    const { getByRole } = render(<Search value="Hello" onChange={handleChange} />);

    fireEvent.change(getByRole("textbox"), { target: { value: "hi" } });

    expect(getByRole("textbox")).toBeInTheDocument();
    expect(handleChange).toBeCalled();
    expect(getByRole("textbox")).toMatchSnapshot();
  });

  test("testing by placeholder of search", () => {
    const { getByPlaceholderText } = render(<Search />);

    expect(getByPlaceholderText("Search...")).toBeInTheDocument();
    expect(getByPlaceholderText("Search...")).toMatchSnapshot();
  });

  test("testing by value of search", () => {
    const { getByDisplayValue } = render(<Search value="Hello" />);

    expect(getByDisplayValue("Hello")).toBeInTheDocument();
    expect(getByDisplayValue("Hello")).toMatchSnapshot();
  });
});
