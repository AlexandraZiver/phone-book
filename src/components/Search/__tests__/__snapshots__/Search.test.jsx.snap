// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Search  testing by onChange of search 1`] = `
<input
  icon="search"
  name="searchInput"
  placeholder="Search..."
  type="text"
  value="Hello"
/>
`;

exports[`testing Search  testing by placeholder of search 1`] = `
<input
  icon="search"
  name="searchInput"
  placeholder="Search..."
  type="text"
  value=""
/>
`;

exports[`testing Search  testing by value of search 1`] = `
<input
  icon="search"
  name="searchInput"
  placeholder="Search..."
  type="text"
  value="Hello"
/>
`;
